core/kernel/console.c

```diff
@@ -126,10 +126,30 @@ void configure_console_from_dt(void)
 	char *parms = NULL;
 	void *fdt;
 	int offs;
+	bool embedded_dt = 1;
+	TEE_Result rc = TEE_ERROR_GENERIC;
 
-	fdt = get_external_dt();
-	if (get_console_node_from_dt(fdt, &offs, &uart, &parms))
-		return;
+	fdt = get_embedded_dt();
+	if (!fdt) {
+		fdt = get_external_dt();
+		if (fdt)
+			embedded_dt = 0;
+		else
+			return;
+	}
+
+	rc = get_console_node_from_dt(fdt, &offs, &uart, &parms);
+	if (rc) {
+		if (embedded_dt == 1) {
+			fdt = get_external_dt();
+			if (get_console_node_from_dt(fdt, &offs, &uart,
+							&parms)) {
+				return;
+			}
+		} else {
+			return;
+		}
+	}
```

Comment:
Maybe a shorter way:
```c
	res = get_console_node_from_dt(get_embedded_dt(),  &offs, &uart, &parms);
	if (res == TEE_ERROR_ITEM_NOT_FOUND)
		res = get_console_node_from_dt(get_external_dt(),  &offs, &uart, &parms);
	if (res) {
		DSMG("No chosen console found");
		return;
	}

Reply:
+1

Reply:
Sorry for the confusion @jenswi-linaro.

There are 2 topics here:
* Embedded DTB versus pager specific _init_ section.
Please see #3834.
* Which DTB to get the chosen from. Here 2 ways:
Either select a DTB and look up console from: @jforissier suggested to use `get_fdt()`.
Or lookup a chosen console in available DTBs: this change you +1'ed.
@jforissier, @jenswi-linaro (and all): your opinion on this DTB selection?